### ISSUE TITLE
typescript/no-var-requires: rule never fires in .js files — `const {x} = require(...)` not flagged, causing false "unused eslint-disable directive"

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25623]
## Summary

`typescript/no-var-requires` fails to flag the canonical CommonJS variable-declaration form (`const {x} = require('node:util')`) in `.js` files, which both ESLint 7 and ESLint 9 with `@typescript-eslint/no-var-requires` flag. The identical constructs **are** flagged when the same file has a `.ts` extension, so the detection logic works — the rule appears to be extension-gated to TypeScript files. Because nothing is reported in `.js`, oxlint (with `--report-unused-disable-directives`) then flags a legitimately-suppressing in-source `eslint-disable` directive as unused, which would cause auto-removal of a directive that real ESLint still needs.

## oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0.

## Reproduction

`repro.js`:

```js
/* eslint-disable @typescript-eslint/no-var-requires */
const {styleText} = require('node:util')
const sdkConfig = require('./configs/sdk.rspack')
```

`.oxlintrc.json`:

```json
{
	"plugins": ["typescript"],
	"categories": {"correctness": "off"},
	"rules": {"typescript/no-var-requires": "error"}
}
```

Run:

```
oxlint -c .oxlintrc.json --report-unused-disable-directives repro.js
```

## Expected

A `typescript/no-var-requires` diagnostic on each `const … = require(...)` line (lines 2 and 3). This is the canonical construct the rule exists to catch: a `require()` call whose parent is a `VariableDeclarator`. ESLint 7.23.0 (`@typescript-eslint/parser` + `@typescript-eslint/eslint-plugin` 6.21.0) and ESLint 9.39.4 (flat config, typescript-eslint 8.62.1) both report `Require statement not part of import statement` on both lines, so the line-1 `eslint-disable` directive is live/used and neither reports it as unused. (The rule is deprecated upstream in typescript-eslint v8 in favor of `no-require-imports`, but still present and functional in 8.62.1.)

## Actual

oxlint 1.78.0 reports no `no-var-requires` diagnostic on the `require()` calls and emits:

```
repro.js:1:1: warning: Unused eslint-disable directive (no problems were reported).
```

With the directive removed: no output, exit 0. Copying the same two lines into a `.ts` file (no directive): both are flagged (`1:21` and `2:19`, `Require statement not part of import statement`) — the rule only ever reports on TypeScript files.

## Cross-checked

| Tool | Result on the two require lines (no directive) |
| --- | --- |
| ESLint 7.23.0 + @typescript-eslint/eslint-plugin 6.21.0 | 2 errors (1:21, 2:19) |
| ESLint 9.39.4 + typescript-eslint 8.62.1 (flat config) | 2 errors (1:21, 2:19) |
| oxlint 1.78.0 `typescript/no-var-requires` | no diagnostics in `.js`; directive reported unused (flagged only if file is `.ts`) |
